website/src/models/group.js

```diff
@@ -352,6 +354,77 @@ GroupSchema.methods.toJSON = function() {
   return doc;
 };
 
+GroupSchema.statics.removeUserFromChallenges = function(user, group, keep, cb) {
```

Comment:
I agree that the name should be more semantic for what is happening in the function.

In addition, let's work on splitting some of these things out into smaller private methods. IE: `_removeUserFromQuest`, `_removeUserFromChallenges`, `_removeUserFromGuilds`, etc. Then, in the master function, it'd be something like:

```js
GroupSchema.statics.removeUserFromAll = function(user, group, keep, cb) {
  async.parallel([
    _removeUserFromQuest(user, group, cb),
    _removeUserFromChallenges(user, group, cb),
    _removeUserFromGuilds(user, group, keep, cb),
    // ... and so on
  ]);
}
```

This will allow individual methods to be small and more easily maintainable. 

I eventually want similar work to happen in all the express controllers.